src/reduce.js

```diff
@@ -7,12 +7,18 @@
  * @returns {*}
  */
 function reduce(callback, startValue) {
+  if (typeof callback !== 'function') {
+    throw new Error('Callback should be a function');
+  }
+
   let prev = startValue;
   let startIndex = 0;
 
-  if (arguments.length < 2) {
+  if (arguments.length < 2 && this[0]) {
     startIndex = 1;
     prev = this[0];
+  } else if (arguments.length < 2 && !this[0]) {
+    throw new Error('Array must not be empty or InitValue should be specified');
   }
 
   for (let i = startIndex; i < this.length; i++) {
```

Comment:
The loop uses `this.length`, which will not work correctly because `this` does not refer to the array. Consider passing the array as an argument and using it directly.